server/api.go

```diff
@@ -52,10 +52,10 @@ func (p *Plugin) initializeAPI() {
 
 	apiRouter.HandleFunc("/user", p.checkAuth(p.attachContext(p.getGitlabUser), ResponseTypeJSON)).Methods(http.MethodPost)
 	apiRouter.HandleFunc("/todo", p.checkAuth(p.attachUserContext(p.postToDo), ResponseTypeJSON)).Methods(http.MethodPost)
```

Comment:
> * For the '/todo' endpoint, I didn't use the `checkToken` wrapper there since I already put a call to `p.checkAndRefreshToken` in `GetToDo` since at that time I was trying to put the token checks as close as possible to the GitLab client calls. I could instead refactor so all the calls to `p.checkAndRefreshToken` (there are 3 in `server/plugin.go`) are moved out of the `server/plugin.go` methods and instead are placed in `api.go` and `command.go`. Is there a preference?

Sorry I read this but still made my comment above. Since we may be going with the strategy of checking at the top of each function that uses the gitlab client, I think having this in `GetToDo` works well.